sweep/txgenerator.go

```diff
@@ -247,45 +251,54 @@ func createSweepTx(inputs []Input, outputPkScript []byte,
 }
 
 // getInputWitnessSizeUpperBound returns the maximum length of the witness for
-// the given input if it would be included in a tx.
-func getInputWitnessSizeUpperBound(input Input) (int, error) {
+// the given input if it would be included in a tx. We also return if the
+// output itself is a nested p2sh output, if so then we need to take into
+// account the extra sigScript data size.
+func getInputWitnessSizeUpperBound(input Input) (int, bool, error) {
 	switch input.WitnessType() {
 
-	// Outputs on a remote commitment transaction that pay directly
-	// to us.
+	// Outputs on a remote commitment transaction that pay directly to us.
+	case lnwallet.WitnessKeyHash:
+		fallthrough
 	case lnwallet.CommitmentNoDelay:
-		return lnwallet.P2WKHWitnessSize, nil
+		return lnwallet.P2WKHWitnessSize, false, nil
```

Comment:
Yeah, would be nice to avoid to have to do this if possible. Instead pass the `Input` directly to the weight estimator.

Reply:
That requires a new package and several updates throughout `lnd`. I think that can stand alone as its own PR instead of expanding this PR further. Once this is in, I can pick up on @joostjager's partial branch to make the move over.